tests/conftest.py

```diff
@@ -37,16 +37,78 @@ def proofs_for_tree_data(tree_data):
     return proofs
 
 
+@pytest.fixture(scope="session")
+def eligible_address_0(tree_data):
+    return tree_data[0].address
+
+
+@pytest.fixture(scope="session")
+def eligible_value_0(tree_data):
+    return tree_data[0].value
+
+
+@pytest.fixture(scope="session")
+def proof_0(proofs_for_tree_data):
+    return proofs_for_tree_data[0]
+
+
 @pytest.fixture(scope="session")
 def root_hash_for_tree_data(tree_data):
     tree = build_tree(tree_data)
     return tree.root.hash
 
 
 @pytest.fixture(scope="session")
-def merkle_drop_contract(deploy_contract, web3, root_hash_for_tree_data):
+def premint_token_owner(accounts):
+    return accounts[0]
+
+
+@pytest.fixture(scope="session")
+def premint_token_value():
+    # The returned value should be higher than the dropped value: see values in `tree_data`
+    return 1000
+
+
+@pytest.fixture(scope="session")
+def dropped_token_contract(
+    deploy_contract, root_hash_for_tree_data, premint_token_owner, premint_token_value
+):
+    # A token contract with premint token for the merkle drop.
+    # The tokens are transferred to the MerkleDrop upon deployment of MerkleDrop.
+    contract = deploy_contract(
+        "DroppedToken",
+        constructor_args=(
+            "droppedToken",
+            "DTN",
+            18,
+            premint_token_owner,
+            premint_token_value,
+        ),
+    )
+
+    return contract
+
+
+@pytest.fixture(scope="session")
```

Comment:
Are you sure that you don't want to have a fresh contract for each test case? We also haven't done so in the tests for the `blockchain` repository. :thinking:

Reply:
Yes, we have done so in the blockchain repository, we have both session scoped initialized and non-initialized contracts. Here I feel like I am not testing the token so I am fine with starting tests with a merkle_drop_contract already owning tokens.